bin/console

```diff
@@ -8,63 +8,56 @@ use Shopware\Production\HttpKernel;
-    exit(1);
-}
-
-set_time_limit(0);
-
-$classLoader = require __DIR__ . '/../vendor/autoload.php';
+$envFile = __DIR__ . '/../.env';
 
-if (!class_exists(Application::class)) {
-    throw new RuntimeException('You need to add "symfony/framework-bundle" as a Composer dependency.');
+if (!file_exists(__DIR__ . '/../.env')) {
+    $_SERVER['APP_RUNTIME_OPTIONS']['disable_dotenv'] = true;
 }
 
-$envFile = __DIR__ . '/../.env';
+require_once __DIR__ . '/../vendor/autoload_runtime.php';
 
-if (class_exists(Dotenv::class) && is_readable($envFile) && !is_dir($envFile)) {
-    (new Dotenv())->usePutenv()->load($envFile);
-}
+return function (array $context) {
+    if (\PHP_VERSION_ID < 70400) {
+        echo 'Your cli is running PHP version ' . \PHP_VERSION . ' but Shopware 6 requires at least PHP 7.4.0' . \PHP_EOL;
+        exit(1);
+    }
 
-if (!isset($_SERVER['PROJECT_ROOT'])) {
-    $_SERVER['PROJECT_ROOT'] = dirname(__DIR__);
-}
+    set_time_limit(0);
 
-$input = new ArgvInput();
-$env = $input->getParameterOption(['--env', '-e'], $_SERVER['APP_ENV'] ?? 'prod', true);
-$debug = ($_SERVER['APP_DEBUG'] ?? ($env !== 'prod')) && !$input->hasParameterOption('--no-debug', true);
+    $classLoader = require __DIR__ . '/../vendor/autoload.php';
 
-if ($debug) {
-    umask(0000);
+    if (!class_exists(Application::class)) {
+        throw new RuntimeException('You need to add "symfony/framework-bundle" as a Composer dependency.');
+    }
 
-    if (class_exists(Debug::class)) {
-        Debug::enable();
+    if (!isset($context['PROJECT_ROOT'])) {
+        $context['PROJECT_ROOT'] = dirname(__DIR__);
     }
-}
 
-$pluginLoader = new StaticKernelPluginLoader($classLoader, null);
+    $input = new ArgvInput();
+    $env = $input->getParameterOption(['--env', '-e'], $context['APP_ENV'] ?? 'prod', true);
+    $debug = ($context['APP_DEBUG'] ?? ($env !== 'prod')) && !$input->hasParameterOption('--no-debug', true);
 
-$shopwareVersion = InstalledVersions::getVersion('shopware/core') . '@' . InstalledVersions::getReference('shopware/core');
+    $pluginLoader = new StaticKernelPluginLoader($classLoader, null);
 
-if ($input->getFirstArgument() === 'system:install') {
-    $_SERVER['INSTALL'] = true;
-}
+    $shopwareVersion = InstalledVersions::getVersion('shopware/core') . '@' . InstalledVersions::getReference('shopware/core');
 
-if (trim($_SERVER['DATABASE_URL'] ?? '') === '') {
-    // fake DATABASE_URL
-    $_SERVER['DATABASE_URL'] = Kernel::PLACEHOLDER_DATABASE_URL;
-} else {
-    if (!isset($_SERVER['INSTALL'])) {
-        $pluginLoader = new DbalKernelPluginLoader($classLoader, null, \Shopware\Core\Kernel::getConnection());
+    if ($input->getFirstArgument() === 'system:install') {
+        $context['INSTALL'] = true;
+    }
+
+    if (trim($context['DATABASE_URL'] ?? '') === '') {
+        // fake DATABASE_URL
+        putenv('DATABASE_URL=' . Kernel::PLACEHOLDER_DATABASE_URL);
```

Comment:
No super-globals anymore. But writing to `$context` would not have any effect. So I use `putenv` here.

Reply:
We don't use putenv / getenv anymore